src/logical/execution/selection_strategy/strategy.rs

```diff
@@ -1,12 +1,27 @@
 //! Contains the trait that defines what constitutes a rule execution strategy.
 
+use thiserror::Error;
+
 use crate::logical::{model::Rule, program_analysis::analysis::RuleAnalysis};
 
+/// Errors that can occur while creating a strategy.
+#[derive(Error, Debug, Copy, Clone)]
+pub enum SelectionStrategyError {
+    /// Non-Stratifyable
+    #[error("The rules of the program cannot be stratified.")]
+    NonStratifiableProgram,
```

Comment:
The wording suggests that we're talking about something like effectively stratifiable programs – but we're really just supporting stratified programs. I would suggest to have “is not stratified” and `NonStratifiedProgram`.